src/Pipeline.h

```diff
@@ -102,38 +100,28 @@ class Pipeline {
         halide_buffer_t *buf{nullptr};
         std::unique_ptr<std::vector<Buffer<>>> buffer_list;
 
-        RealizationArg(Realization &r)
-            : r(&r) {
-        }
-        RealizationArg(Realization &&r)
-            : r(&r) {
-        }
-        RealizationArg(halide_buffer_t *buf)
-            : buf(buf) {
-        }
+        RealizationArg(Realization &r);
+        RealizationArg(Realization &&r);
+        RealizationArg(halide_buffer_t *buf);
+        RealizationArg(RealizationArg &&from);
+
         template<typename T, int D>
-        RealizationArg(Runtime::Buffer<T, D> &dst)
-            : buf(dst.raw_buffer()) {
+        HALIDE_ALWAYS_INLINE RealizationArg(Runtime::Buffer<T, D> &dst)
+            : RealizationArg(dst.raw_buffer()) {
         }
+
         template<typename T>
-        HALIDE_NO_USER_CODE_INLINE RealizationArg(Buffer<T> &dst)
```

Comment:
Not sure if this one can throw an error or not. We can only use ALWAYS_INLINE if it won't.